deployment/clouddeploy/deployer.go

```diff
@@ -357,18 +468,161 @@ func (p *Deployer) buildModifySpecs(
 	return specs, nil
 }
 
-func (p *Deployer) NewCluster(ctx context.Context, def *clusterdef.Cluster) (deployment.ClusterInfo, error) {
-	var clusterVersion string
-	for _, nodeGroup := range def.NodeGroups {
-		if clusterVersion == "" {
-			clusterVersion = nodeGroup.Version
+func (p *Deployer) DeployNewCluster(ctx context.Context, def *clusterdef.Cluster, clusterVersion string, serverImage string) (deployment.ClusterInfo, error) {
```

Comment:
This should probably be private rather than public I would think (`deployeNewCluster`).

Reply:
Certainly, I was not aware that the visibility of functions is determined by the case of their first letter until now. Haven't much worked on go yet.